src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs

```diff
@@ -504,17 +503,16 @@ private static void ImmediateCancelText()
                     StringBuilder sb = new StringBuilder();
                     for (int i = 0; i < 1000000; i++)
                         sb.Append(i);
-                    cmd.CommandText = "insert into #blobs (Id, blob) values (1, @blob)";
+                    // Include a delay to allow time for cancellation
+                    cmd.CommandText = "WAITFOR DELAY '00:00:05'; insert into #blobs (Id, blob) values (1, @blob)";
```

Comment:
Curious about the delay time. How did we end up to that exact number of 5 seconds? Is it based on experience or a fact or I am missing something?

Reply:
It's based on experience; you can find it in another test in the same class that it's already running well.

Reply:
Can we avoid this delay or reduce it to 1-2 secs? It eventually adds up to test execution time 👀

Reply:
These tests are supposed to be canceled immediately, and this waiting time is just for safety. I'm afraid reducing them may increase the chance of failure.